src/Microsoft.AspNetCore.Razor/CodeGenerators/Visitors/CSharpTagHelperFieldDeclarationVisitor.cs

```diff
@@ -168,6 +188,23 @@ private void PreAllocateUnboundTagHelperAttributes(TagHelperChunk chunk)
             }
         }
 
+        private static bool IsLiteralAttributeValue(Chunk attributeValueChunk)
```

Comment:
`ExpressionChunk`s are typically due to `@Foo` in an attribute. If you point me to where you're seeing them I can clarify further.